From 8545bfe31a4acde4d20273f4abf18587dc55144c Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Mon, 9 Sep 2024 03:40:26 -0400
Subject: [PATCH] Revert "Add workaround for several i3c services (#288)
 (#305)"

-- reverted the commit to fix mctp dependencies build error (not found mctpPtr->getServiceName())

This reverts commit 5b8abe5ffaed7d9d37cbc03be154482db733d179.
---
 src/pmt.cpp             | 82 ++++++++++------------------------
 src/pmt.hpp             |  4 +-
 subprojects/libpmt.wrap |  2 +-
 tests/src/pmt_test.cpp  | 97 ++++-------------------------------------
 4 files changed, 34 insertions(+), 151 deletions(-)

diff --git a/src/pmt.cpp b/src/pmt.cpp
index 5777b93..bba96fa 100644
--- a/src/pmt.cpp
+++ b/src/pmt.cpp
@@ -39,7 +39,6 @@
 #include <metrics/metric_sink.hpp>
 #include <metrics/metrics_monitor.hpp>
 #include <pmt/logger.hpp>
-#include <pmt/transport/mctp/endpoint_iface.hpp>
 #include <pmt/transport/transport.hpp>
 #include <pmt/utils.hpp>
 #include <sdbusplus/asio/property.hpp>
@@ -549,58 +548,35 @@ inline bool Pmt::shouldConfigureWatcher(const Watcher& watcher) const
     return true;
 }
 
-std::optional<mctp_eid_t>
-    Pmt::getBmcEidForEndpoint(std::shared_ptr<PmtEndpointIface> endpoint) const
-{
-    if (auto mctpPtr =
-            std::dynamic_pointer_cast<transport::mctp::EndpointIface>(
-                endpoint->getAddress());
-        bool(mctpPtr))
-    {
-        if (auto it = eids.find(mctpPtr->getServiceName()); it != eids.end())
-        {
-            return it->second;
-        }
-    }
-    else
-    {
-        LogMsg(Logger::Debug, "Endpoint ", endpoint->getAddress()->str(),
-               " is not MCTP type. Couldn't get BMC EID.");
-    }
-    return std::nullopt;
-}
-
 Status Pmt::discoverTokenId(boost::asio::yield_context yield,
                             std::shared_ptr<PmtEndpointIface> endpoint)
 {
     if (endpoint->needsTokenIdDiscovery())
     {
-        if (auto eid = getBmcEidForEndpoint(endpoint); eid)
-        {
-            uint8_t tokenId = 0;
-            auto status = discovery::watcher::TokenIdDiscovery(
-                              *transport, endpoint->getAddress())
-                              .configureTokenId(yield, tokenId, *eid);
+        uint8_t tokenId = 0;
+        auto status = discovery::watcher::TokenIdDiscovery(
+                          *transport, endpoint->getAddress())
+                          .configureTokenId(yield, tokenId, *eid);
 
-            if (status != Status::success)
-            {
-                LogMsg(Logger::Debug,
-                       "ConfigureTokenId status: ", utils::toString(status),
-                       " for endpoint: ", endpoint->getAddress()->shortStr());
-                return status;
-            }
-            endpoint->markTokenIdDiscovered(tokenId);
-        }
-        else
+        if (status != Status::success)
         {
-            return Status::failure;
+            LogMsg(Logger::Debug,
+                   "ConfigureTokenId status: ", utils::toString(status),
+                   " for endpoint: ", endpoint->getAddress()->shortStr());
+            return status;
         }
+        endpoint->markTokenIdDiscovered(tokenId);
     }
     return Status::success;
 }
 
 void Pmt::configureWatchers(boost::asio::yield_context yield)
 {
+    if (!eid)
+    {
+        return;
+    }
+
     for (const auto& endpoint : getEndpoints())
     {
         Status status = discoverTokenId(yield, endpoint);
@@ -818,31 +794,20 @@ const std::vector<std::shared_ptr<PmtEndpointIface>> Pmt::getEndpoints() const
 
 void Pmt::configureWatcherLoop(boost::asio::yield_context yield)
 {
-    transport->getOwnEIDs([this](mctp_eid_t newEid, std::string service) {
+    transport->getOwnEIDs([this](mctp_eid_t newEid) {
         if (newEid == 0)
         {
-            eids.erase(service);
+            eid = std::nullopt;
             LogMsg(Logger::Debug,
                    "Received Eid equal to 0. Assuming it is invalid.");
             return;
         }
-        LogMsg(Logger::Info,
-               std::format("Received Eid equal to {} from service: {}", newEid,
-                           service));
-        eids[service] = newEid;
+
+        eid = newEid;
 
         for (const auto& endpoint : getEndpoints())
         {
-            if (auto mctpPtr =
-                    std::dynamic_pointer_cast<transport::mctp::EndpointIface>(
-                        endpoint->getAddress());
-                bool(mctpPtr))
-            {
-                if (mctpPtr->getServiceName() == service)
-                {
-                    endpoint->markTokenIdNotDiscovered();
-                }
-            }
+            endpoint->markTokenIdNotDiscovered();
         }
     });
 
@@ -887,6 +852,7 @@ void Pmt::monitoringLoop(boost::asio::yield_context yield)
 inline autogen::DataSources chooseSourceForMetric(
     const std::vector<autogen::DataSources>& redfishSensorInputs)
 {
+
     if (std::ranges::contains(redfishSensorInputs,
                               autogen::DataSources::ThresholdWatcher))
     {
@@ -1003,8 +969,7 @@ void Pmt::observe(const std::shared_ptr<PmtEndpointIface>& endpoint)
                 if (groupIt == schema->groups.end())
                 {
                     LogMsg(Logger::Error,
-                           "Inconsistency in schema. Unable to find "
-                           "group ",
+                           "Inconsistency in schema. Unable to find group ",
                            ingredient.groupIDREF);
                     continue;
                 }
@@ -1014,8 +979,7 @@ void Pmt::observe(const std::shared_ptr<PmtEndpointIface>& endpoint)
                 if (sampleIt == group.samples.end())
                 {
                     LogMsg(Logger::Error,
-                           "Inconsistency in schema. Unable to find "
-                           "sample ",
+                           "Inconsistency in schema. Unable to find sample ",
                            ingredient.sampleIDREF, " within group ",
                            ingredient.groupIDREF);
                     continue;
diff --git a/src/pmt.hpp b/src/pmt.hpp
index fb98264..9cfa11c 100644
--- a/src/pmt.hpp
+++ b/src/pmt.hpp
@@ -119,7 +119,7 @@ class Pmt
     void observe(const std::shared_ptr<PmtEndpointIface>& endpoint);
     void stopPmt();
 
-    std::map<std::string, mctp_eid_t> eids;
+    std::optional<mctp_eid_t> eid;
 
   protected:
     void monitoringLoop(boost::asio::yield_context yield);
@@ -139,8 +139,6 @@ class Pmt
     void configureWatcherLoop(boost::asio::yield_context yield);
     bool shouldStopStream(const Watcher& watcher) const;
     bool shouldConfigureWatcher(const Watcher& watcher) const;
-    std::optional<mctp_eid_t>
-        getBmcEidForEndpoint(std::shared_ptr<PmtEndpointIface> endpoint) const;
     Status discoverTokenId(boost::asio::yield_context yield,
                            std::shared_ptr<PmtEndpointIface> endpoint);
     void configureStreamingWatcher(boost::asio::yield_context yield,
diff --git a/subprojects/libpmt.wrap b/subprojects/libpmt.wrap
index b3883d5..8426b52 100644
--- a/subprojects/libpmt.wrap
+++ b/subprojects/libpmt.wrap
@@ -1,6 +1,6 @@
 [wrap-git]
 url = https://github.com/intel-collab/firmware.bmc.openbmc.libraries.libpmt.git
-revision = 2e366c68b66dfe992eceeb2bf3a0df94c4605843
+revision = 42486093eb0f0212a6d2ef9236ea146672a13025
 
 [provide]
 libpmt = libpmt_dep
diff --git a/tests/src/pmt_test.cpp b/tests/src/pmt_test.cpp
index 2f4bcb2..c54d110 100644
--- a/tests/src/pmt_test.cpp
+++ b/tests/src/pmt_test.cpp
@@ -97,15 +97,12 @@ class PmtTests : public Test
 
     std::vector<std::shared_ptr<transport::EndpointIface>> addresses{
         endpointPtr};
-    std::function<void(mctp_eid_t, std::string)> getOwnEidsCallback;
-    static constexpr std::string defaultMctpServiceName = "service0";
+    std::function<void(mctp_eid_t)> getOwnEidsCallback;
     const mctp_eid_t pmtEid = 0x09;
 
     PmtTests()
     {
         ON_CALL(*endpointPtr, isActive()).WillByDefault(Return(true));
-        ON_CALL(*endpointPtr, getServiceName())
-            .WillByDefault(Return(defaultMctpServiceName));
         ON_CALL(transportMock, getOwnEIDs(_))
             .WillByDefault(SaveArg<0>(&getOwnEidsCallback));
 
@@ -117,12 +114,10 @@ class PmtTests : public Test
             }));
     }
 
-    void
-        callGetOwnEIDsCallback(mctp_eid_t eid,
-                               std::string serviceName = defaultMctpServiceName)
+    void callGetOwnEIDsCallback(mctp_eid_t eid)
     {
         ASSERT_TRUE(static_cast<bool>(getOwnEidsCallback));
-        getOwnEidsCallback(eid, serviceName);
+        getOwnEidsCallback(eid);
     }
 
     void removeEndpoint(boost::asio::yield_context yield,
@@ -191,42 +186,6 @@ TEST_F(PmtTests, PmtInitializeInitiatesDiscovery)
     initializePmt();
 }
 
-TEST_F(PmtTests, EidEqualZeroClearsMappingForService)
-{
-    initializePmt();
-
-    callGetOwnEIDsCallback(7, "service21");
-    callGetOwnEIDsCallback(0, "service21");
-
-    ASSERT_EQ(0, pmt->eids.size());
-}
-
-TEST_F(PmtTests, GetOwnEidsCallbackCreatesCorrectMapping)
-{
-    initializePmt();
-
-    callGetOwnEIDsCallback(5, "service21");
-    callGetOwnEIDsCallback(7, "service37");
-
-    ASSERT_EQ(2, pmt->eids.size());
-    EXPECT_EQ(5, pmt->eids["service21"]);
-    EXPECT_EQ(7, pmt->eids["service37"]);
-}
-
-TEST_F(PmtTests, GetOwnEidsCallbackUpdatesExistingMapping)
-{
-    initializePmt();
-
-    callGetOwnEIDsCallback(5, "service21");
-    callGetOwnEIDsCallback(7, "service37");
-
-    callGetOwnEIDsCallback(9, "service21");
-
-    ASSERT_EQ(2, pmt->eids.size());
-    EXPECT_EQ(9, pmt->eids["service21"]);
-    EXPECT_EQ(7, pmt->eids["service37"]);
-}
-
 // TODO: When async callback will be implemented change the
 // implementation to wait until discovery is done instead of using
 // timeout.
@@ -820,7 +779,7 @@ class PmtConfigureStreamTests : public PmtTests
     {
         PmtTests::SetUp();
         setAggregatorSchemas();
-        this->pmt->eids[defaultMctpServiceName] = pmtEid;
+        this->pmt->eid = pmtEid;
 
         using transport::api::WatcherType;
         transport::WatcherDetails watchersDetails = {.watcherType =
@@ -853,7 +812,8 @@ class PmtConfigureStreamTests : public PmtTests
         ON_CALL(this->transportMock,
                 getAgentParameter(An<boost::asio::yield_context>(),
                                   Eq(this->endpointPtr), _, _, _))
-            .WillByDefault(Return(std::pair(Status::success, pmtEid)));
+            .WillByDefault(
+                Return(std::pair(Status::success, *(this->pmt->eid))));
         ON_CALL(this->transportMock,
                 getWatcherConfiguration(An<boost::asio::yield_context>(), _, _,
                                         static_cast<uint64_t>(0)))
@@ -1078,17 +1038,6 @@ TEST_F(PmtConfigureStreamTests, NoStreamingDestinationDoesNotConfigure)
     this->initializePmt();
 }
 
-TEST_F(PmtConfigureStreamTests, NoMctpServiceNameMappingDoesNotConfigure)
-{
-    this->pmt->eids.clear();
-
-    expectTurnOffStream().Times(0);
-    expectSetWatcherConfigurationBulk().Times(0);
-    expectSetWatcherConfiguration().Times(0);
-
-    this->initializePmt();
-}
-
 TEST_F(PmtConfigureStreamTests, EndpointRemovedDuringGetAgentParameter)
 {
 
@@ -1292,7 +1241,7 @@ class PmtConfigureTracerWatcherTests : public PmtTests
             configValues.pushIntervalMs(), configValues.mctpTimeoutMs(), true,
             true);
         PmtTests::SetUp();
-        this->pmt->eids[defaultMctpServiceName] = pmtEid;
+        this->pmt->eid = pmtEid;
 
         using transport::api::WatcherType;
         transport::WatcherDetails watchersDetails = {.watcherType =
@@ -1337,7 +1286,8 @@ class PmtConfigureTracerWatcherTests : public PmtTests
         ON_CALL(this->transportMock,
                 getAgentParameter(An<boost::asio::yield_context>(),
                                   Eq(this->endpointPtr), _, _, _))
-            .WillByDefault(Return(std::pair(Status::success, pmtEid)));
+            .WillByDefault(
+                Return(std::pair(Status::success, *(this->pmt->eid))));
         ON_CALL(this->transportMock,
                 getSampleGroup(An<boost::asio::yield_context>(), _, _, _, _, _))
             .WillByDefault(Return(
@@ -1834,8 +1784,6 @@ class PmtTestsWithEndpointMock : public PmtTests
             std::make_shared<NiceMock<transport::mctp::EndpointMock>>();
 
         ON_CALL(*mctpEndpointMock, isActive()).WillByDefault(Return(true));
-        ON_CALL(*mctpEndpointMock, getServiceName())
-            .WillByDefault(Return(defaultMctpServiceName));
 
         auto pmtEndpointMock = std::make_shared<NiceMock<PmtEndpointMock>>();
 
@@ -1944,33 +1892,6 @@ TEST_F(PmtTestsWithEndpointMock,
     ioc.run_for(configureWatchersRetryInterval + configureWatchersEpsilon);
 }
 
-TEST_F(PmtTestsWithEndpointMock, EndpointsFromDifferentNetworksMarkOnlyCorrect)
-{
-    auto [pmtEndpointMock1, mctpEndpointMock1] = makeEndpointMock();
-    auto [pmtEndpointMock2, mctpEndpointMock2] = makeEndpointMock();
-    ON_CALL(*mctpEndpointMock2, getServiceName).WillByDefault(Return("mctp2"));
-
-    expectGetStreamAgentCount(mctpEndpointMock1)
-        .WillRepeatedly(Return(std::make_pair(Status::success, 5)));
-    expectGetStreamAgentCount(mctpEndpointMock2)
-        .WillRepeatedly(Return(std::make_pair(Status::success, 5)));
-
-    expectGetAgentMctpEid(mctpEndpointMock1)
-        .WillRepeatedly(returnEid(dummyEid));
-    expectGetAgentMctpEid(mctpEndpointMock2)
-        .WillRepeatedly(returnEid(dummyEid + 1));
-
-    EXPECT_CALL(*pmtEndpointMock1, markTokenIdNotDiscovered).Times(0);
-    EXPECT_CALL(*pmtEndpointMock2, markTokenIdNotDiscovered());
-    EXPECT_CALL(*pmtEndpointMock2, markTokenIdDiscovered(0));
-
-    this->initializePmt();
-
-    callGetOwnEIDsCallback(dummyEid + 1, "mctp2");
-
-    ioc.run_for(configureWatchersRetryInterval + configureWatchersEpsilon);
-}
-
 class PmtMonitoringTests : public Test
 {
   public:
-- 
2.34.1

